uniffi_core/src/ffi_converter_traits.rs

```diff
@@ -247,6 +247,7 @@ pub unsafe trait Lower<UT>: Sized {
     fn lower_into_rust_buffer(obj: Self) -> RustBuffer {
         let mut buf = ::std::vec::Vec::new();
         Self::write(obj, &mut buf);
+        buf.shrink_to(i32::MAX as usize);
```

Comment:
This line is what fixed our issue downstream. I patched the other places `RustBuffer::from_vec` was being called in case that was wanted, but happy to remove them if not.

Reply:
I agree that it would be good to avoid an extra copy here.  What about changing the `write()` implementations to call `reserve_exact` before pushing to the Vec?  Unfortunately, I think that means implementing our own growth algorithm, but something simple should work okay.